enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java

```diff
@@ -134,17 +147,34 @@ public boolean handleRequest(RequestContext requestContext) {
             return true;
         }
 
+        boolean authenticated = false;
         boolean canAuthenticated = false;
         for (Authenticator authenticator : authenticators) {
             if (authenticator.canAuthenticate(requestContext)) {
-                canAuthenticated = true;
+                if (!authenticator.getName().contains("mtls")) {
+                    canAuthenticated = true;
+                }
                 AuthenticationResponse authenticateResponse = authenticate(authenticator, requestContext);
-                if (authenticateResponse.isAuthenticated() && !authenticateResponse.isContinueToNextAuthenticator()) {
+                if (authenticateResponse.isMandatoryAuthentication()) {
+                    authenticated = authenticateResponse.isAuthenticated();
                     setInterceptorAuthContextMetadata(authenticator, requestContext);
-                    return true;
+                }
+                if (!authenticateResponse.isContinueToNextAuthenticator()) {
+                    break;
+                }
+            } else {
+                if (isMutualSSLMandatory && authenticator.getName().contains("mtls")) {
+                    log.error("MutualSSL: Failed");
```

Comment:
Maybe we can convert this to a debug log. Should add API name and other details.

Reply:
Converted